### PR TITLE
chore: add APM integration to CI

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It adds a parallel execution running the tests for the APM Integration. Besides that, because this PR was outdated with the codebase, I'm doing the following tasks:

- update the test scenarios with latest versions of the steps
- remove fleet server bootstrapping from the stand-alone scenarios, as it's already provided by the runtime dependencies (starting the Fleet test suite implies starting the stack + fleet-server).
- decouple operations for Fleet integrations into a separate feature file. This way we will be able to differentiate potential errors related to CRUD operations in the integrations from errors. For that, we are moving the operations to the new feature file, removing them from the original scenarios. Therefore, these scenarios will be simpler, testing one specific behaviour.
- minor refactors related to removing leftovers or renaming scenarios for consistency
- **properly enroll standalone agent into Fleet (cc/ @EricDavisX @adam-stokes @michalpristas)**: we discovered that the standalone agent never enrolled in Fleet, and the `the stand-alone agent is listed in Fleet as "online"` step was getting the online status for the wrong agent (the first agent in the list, which was the already bootstrapped fleet-server agent). For that reason we are getting the hostname of current agent and checking that it appears in the response of the agents list. If so, we verify that it's in the online status. The auto-enrollment has been implemented with ce45452e138466de4d8ffbe7ecafefeeb8c54710, where the proper env vars have been declared and populated so the agent enrolls in Fleet using the already bootstrapped fleet-server.

## Why is it important?
We were not running those tests on CI

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have run the Unit tests for the CLI, and they are passing locally
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [x] @jalvz, we are running the tests in parallel on CI, can you confirm that this is what we want for those tests?
- [x] @elastic/apm-server as discussed offline, we'd need your help in fixing these tests.

## How to test this PR locally
```shell
TAGS="apm_server && install" TIMEOUT_FACTOR=5 LOG_LEVEL=TRACE DEVELOPER_MODE=true ELASTIC_APM_ACTIVE=false make -C e2e/_suites/fleet functional-test
```

## Related issues

- Completes #1083

## Backports
Backport to 7.x, and check if 7.14.x and 7.13.x are required

### CODE REVIEW

**Comment by mdelapenya** [comment]
Ok the APM tests are failing on CI. @jalvz could you take a look before merging this PR?

**Review by jalvz — COMMENTED** [review]
oh i didn't know new tests did not run on CI... do we need also the cloud tag?

**Comment by mdelapenya** [comment]
> oh i didn't know new tests did not run on CI... do we need also the cloud tag?

No, because the `@apm_server` tag defines the scope for the entire feature file. We usually use that feature tag for splitting scenarios on CI, using the descriptor modified in this PR

**Review by cachedout — APPROVED** [review]

**Comment by mdelapenya** [comment]
@adam-stokes I'm currently working in this branch, as there are some outdated scenarios that need rework

**Comment by mdelapenya** [comment]
@jalvz sorry for taking that much in pushing this forward. I've just returned from vacations and would like to merge this ASAP.

I've found one issue with the current implementation of the `aStandaloneAgentIsDeployed` test method: it never achieves the `apm-server` process to be spun up in the agent. On the contrary, mounting the config files for "apm-legacy" as a volume (exactly the same as in the `cloud` scenario), then the process appears in the agent container. And that effect would lead to having exactly two equal scenarios.

Do you know why?

**Comment by mdelapenya** [comment]
Updated the description and pinged @elastic/apm-server team so that they are aware of how to reproduce this failure for the APM integration. As mentioned on Slack, they are testing the Fleet integration on every APM Server's PR, so they pointed out to a possible problem in tests' setup.

**Comment by mdelapenya** [comment]
Thousand thank yous to @michalpristas and @axw for the quick chats we had to unblock this.

image

File: e2e/_suites/fleet/features/stand_alone_agent.feature
Comment (mdelapenya) [inline e2e/_suites/fleet/features/stand_alone_agent.feature:10]: This was an error hidden by the non-autoenrolling stand-alone agent.

**Pull request merged**
